prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -458,7 +490,7 @@ class ActionHistory {
         var parentReports2 = mutableSetOf<ReportId>()
         parentReports2.addAll(reportsReceived.keys)
         parentReports2.addAll(reportsIn.keys)
-        val childReports2 = reportsOut.keys
+        val childReports2 = reportsOut.filterValues { it.itemCount > 0 }.keys
```

Comment:
How do we end up with something in reportsOut that is not in fact a child report?